### PR TITLE
TST: Test against pysat v3.0.1RC

### PR DESCRIPTION
**Pull request by rstoneback**
# Description

Addresses # (issue)

 - Original:Testing against pysat v3.0.1RC
 - Now: Updates against latest pysat release version. Removes any temporary tweaks for RC check.
 - Testing expanded to more OS platforms!

# Type of change

Please delete options that are not relevant.

- Release Candidate Testing

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide
instructions so we can reproduce the problem and the solution. Including images
or test files is frequently very useful.  Please also list any relevant details
for your test configuration.

Testing against package as is.

## Test Configuration
* Operating system: GitHub Actions
* Version number: Python 3.7, 3.8, 3.9
* Any details about your local setup that are relevant

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

If this is a release PR, replace the first item of the above checklist with the
release checklist on the pysat wiki:
https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

**Comment by rstoneback** [comment]
Test coverage went down on some ftp based data sources. Not sure what that is about.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -11,8 +11,8 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-latest]
-        python-version: ["3.8", "3.9"]
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: [3.8, 3.9]
```
Comment (rstoneback) [inline .github/workflows/main.yml:15]:
```suggestion
        python-version: ["3.8", "3.9"]
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -11,12 +11,12 @@ jobs:
...
       fail-fast: false
       matrix:
-        os: [ubuntu-latest]
+        os: [ubuntu-latest, macos-latest, windows-latest]
         python-version: ["3.8", "3.9"]
         numpy_ver: [latest]
         include:
           - python-version: "3.7"
-            numpy_ver: "1.17"
+            numpy_ver: "1.17.3"
```
Comment (jklenzing) [inline .github/workflows/main.yml:19]:
```suggestion
            numpy_ver: "1.18"
```

**Review by jklenzing — APPROVED** [review]
Looks good.  Thanks @rstoneback!

**Comment by jklenzing** [comment]
It's just a change to the yml, I'll pull in the latest develop changes to make sure it still works.

**Comment by jklenzing** [comment]
Screen Shot 2021-10-08 at 1 27 11 PM
Actually, looks like it's safe to merge.

**Pull request merged**